### PR TITLE
Clarify block_reduce func keyword parameter

### PR DESCRIPTION
**Pull request by larrybradley**
### Description
This PR clarifies the `block_reduce` `func` keyword parameter based on a question from a user.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

File: astropy/nddata/blocks.py [hunk 1]
```diff
@@ -113,12 +113,12 @@ def block_reduce(data, block_size, func=np.sum):
...
 
     func : callable, optional
-        The method to use to downsample the data.  Must be a callable
-        that takes in a `~numpy.ndarray` along with an ``axis`` keyword,
-        which defines the axis or axes along which the function is
-        applied.  The ``axis`` keyword must accept multiple axes as a
-        tuple.  The default is `~numpy.sum`, which provides block
-        summation (and conserves the data sum).
+        The method to use to downsample the data. Must be a callable
+        that takes in a 4D `~numpy.ndarray` and has an ``axis``
```
Comment (pllim) [inline astropy/nddata/blocks.py:117]: Must it always be 4D? I thought I can pass in 2D too.

Comment (larrybradley) [reply astropy/nddata/blocks.py:117]: Yes.  The `block_reduce` function takes a 2D array, but reshapes it as 4D (using [`reshape_as_blocks`](https://docs.astropy.org/en/stable/api/astropy.nddata.reshape_as_blocks.html?highlight=reshape_as_blocks)) and then passes the 4D array to the `func` callable with `axis=(2,3)` to get back a reduced 2D array.  Make sense?

Comment (larrybradley) [reply astropy/nddata/blocks.py]: OK, I updated the docstring.

**Review by pllim — APPROVED** [review]
Yes, clear to me now. Thanks!

**Comment by pllim** [comment]
We only need RTD to pass here, so I am canceling the irrelevant CI.

**Comment by pllim** [comment]
Thanks!

**Pull request merged**